Disable AppCache everywhere

szewai opened the pull request:
#### f2ffb0129de49a3c02f2b513bba61c889394fbab
<pre>
Disable AppCache everywhere
<a href="https://bugs.webkit.org/show_bug.cgi?id=259783">https://bugs.webkit.org/show_bug.cgi?id=259783</a>
<a href="https://rdar.apple.com/113343269">rdar://113343269</a>

Reviewed by Sihui Liu.

We remove the offline app cache runtime flag.
We disable using app cache, although we still expose DOMApplicationCache API based on the existing quirk.
We remove the tests and a follow-up will remove the implementation.

* LayoutTests/TestExpectations:
* LayoutTests/fast/dom/dom-window-property-gc-after-frame-detach-expected.txt:
* LayoutTests/fast/dom/dom-window-property-gc-after-frame-detach.html:
* LayoutTests/fast/dom/event-handler-attributes.html:
* LayoutTests/fast/events/destroyed-atomic-string.html:
* LayoutTests/fast/events/dispatchEvent-crash-expected.txt:
* LayoutTests/fast/events/dispatchEvent-crash.html:
* LayoutTests/fast/loader/window-properties-restored-from-page-cache-expected.txt:
* LayoutTests/fast/loader/window-properties-restored-from-page-cache.html:
* LayoutTests/http/tests/appcache/: Removed.
* LayoutTests/http/tests/contentfiltering/load-substitute-data-from-appcache-expected.txt: Removed.
* LayoutTests/http/tests/contentfiltering/load-substitute-data-from-appcache.html: Removed.
* LayoutTests/http/tests/contentfiltering/resources/appcache.html: Removed.
* LayoutTests/http/tests/dom/cross-origin-detached-window-properties-expected.txt:
* LayoutTests/http/tests/dom/cross-origin-detached-window-properties.html:
* LayoutTests/http/tests/dom/same-origin-detached-window-properties-expected.txt:
* LayoutTests/http/tests/dom/same-origin-detached-window-properties.html:
* LayoutTests/http/tests/security/appcache-in-private-browsing-expected.txt: Removed.
* LayoutTests/http/tests/security/appcache-in-private-browsing.html: Removed.
* LayoutTests/http/tests/security/appcache-switching-private-browsing-expected.txt: Removed.
* LayoutTests/http/tests/security/appcache-switching-private-browsing.html: Removed.
* LayoutTests/http/tests/security/cross-origin-appcache-allowed-expected.txt: Removed.
* LayoutTests/http/tests/security/cross-origin-appcache-allowed.html: Removed.
* LayoutTests/http/tests/security/cross-origin-appcache-expected.txt: Removed.
* LayoutTests/http/tests/security/cross-origin-appcache.html: Removed.
* LayoutTests/http/tests/security/resources/cross-origin-iframe-for-appcache-allowed.html: Removed.
* LayoutTests/http/tests/security/resources/cross-origin-iframe-for-appcache.html: Removed.
* LayoutTests/http/tests/security/resources/frame-for-appcache-in-private-browsing.html: Removed.
* LayoutTests/http/tests/security/resources/same-origin-iframe-for-appcache-blocked.html: Removed.
* LayoutTests/http/tests/security/same-origin-appcache-blocked-expected.txt: Removed.
* LayoutTests/http/tests/security/same-origin-appcache-blocked.html: Removed.
* LayoutTests/http/tests/ssl/resources/origin-usage-iframe-1.html: Removed.
* LayoutTests/http/tests/ssl/resources/origin-usage-iframe-2.html: Removed.
* LayoutTests/http/tests/ssl/upgrade-origin-usage-expected.txt: Removed.
* LayoutTests/http/tests/ssl/upgrade-origin-usage.html: Removed.
* LayoutTests/js/frame-application-cache-with-listener-crash.html: Removed.
* LayoutTests/platform/ios-wk2/TestExpectations:
* LayoutTests/platform/mac-monterey-wk1/TestExpectations:
* LayoutTests/platform/mac-monterey-wk1/http/tests/appcache/remove-cache-expected.txt: Removed.
* LayoutTests/platform/mac-monterey/TestExpectations:
* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/platform/mac-ventura/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/wincairo/TestExpectations:
* LayoutTests/platform/wk2/TestExpectations:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.cpp:
(WTF::computeSDKAlignedBehaviors):
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WebCore/loader/appcache/ApplicationCacheGroup.cpp:
(WebCore::ApplicationCacheGroup::selectCache):
(WebCore::ApplicationCacheGroup::selectCacheWithoutManifestURL):
* Source/WebCore/loader/appcache/ApplicationCacheHost.cpp:
(WebCore::ApplicationCacheHost::isApplicationCacheEnabled):
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldEnableApplicationCacheQuirk const):
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultOfflineWebApplicationCacheEnabled): Deleted.
* Source/WebKit/Shared/WebPreferencesDefaultValues.h:
* Source/WebKit/UIProcess/API/C/WKPreferences.cpp:
(WKPreferencesSetOfflineWebApplicationCacheEnabled):
(WKPreferencesGetOfflineWebApplicationCacheEnabled):
* Source/WebKit/UIProcess/API/C/WKPreferencesRef.h:
* Source/WebKit/UIProcess/API/Cocoa/WKPreferences.mm:
(-[WKPreferences _offlineApplicationCacheIsEnabled]):
(-[WKPreferences _setOfflineApplicationCacheIsEnabled:]):
* Source/WebKit/UIProcess/API/Cocoa/WKPreferencesPrivate.h:
* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webkit_settings_class_init):
(webkit_settings_get_enable_offline_web_application_cache):
(webkit_settings_set_enable_offline_web_application_cache):
* Source/WebKit/UIProcess/API/glib/WebKitSettings.h.in:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_historyItemClient):
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
* Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
(-[WebPreferences offlineWebApplicationCacheEnabled]):
(-[WebPreferences setOfflineWebApplicationCacheEnabled:]):
* Tools/DumpRenderTree/TestOptions.cpp:
(WTR::TestOptions::defaults):
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKit/WKPreferences.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsiteDataStoreCustomPaths.mm:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitSettings.cpp:
(testWebKitSettings):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebsiteData.cpp:
(serverCallback):
(testWebsiteDataConfiguration):
(testWebsiteDataEphemeral):
(beforeAll):
(testWebsiteDataAppcache): Deleted.
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):

Originally-landed-as: 39f334671a68. <a href="https://bugs.webkit.org/show_bug.cgi?id=259783">https://bugs.webkit.org/show_bug.cgi?id=259783</a>
Canonical link: <a href="https://commits.webkit.org/273297@main">https://commits.webkit.org/273297@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/88c9b624a55b6669275f26910c6dc5bf8b889d4c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35005 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/13896 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/37083 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/37754 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/31617 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/36164 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/16281 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/10974 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/30557 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/35548 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/11773 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31218 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/10322 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/35274 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/10379 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/31307 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39005 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/29736 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/31837 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/31636 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/36404 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/34889 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/10483 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/8409 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/34391 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/12284 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/30951 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/41511 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8022 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11016 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/8638 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/11348 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/loader/appcache/DOMApplicationCache.idl:
```diff
@@ -29,8 +29,7 @@
     GenerateIsReachable=ReachableFromDOMWindow,
     InterfaceName=ApplicationCache,
-    Exposed=Window,
-    EnabledByQuirk=shouldEnableApplicationCache
```
Doesn't this end up always enabling this interface? Similar issue below with the `applicationCache` getter.

I think if you remove the quirk that sometimes enable this you can always remove this entire file.

szewai commented on Source/WebKit/Scripts/process-entitlements.sh:
```diff
@@ -175,12 +175,10 @@ function mac_process_network_entitlements()
 function webcontent_sandbox_entitlements()
 {
     plistbuddy Add :com.apple.private.security.mutable-state-flags array
-    plistbuddy Add :com.apple.private.security.mutable-state-flags:0 string AppCacheDisabled
```
I think next index should start with 0 if this is removed. cc @pvollan to confirm

youennf replied:
Will update

szewai commented on Source/WebCore/loader/appcache/ApplicationCacheGroup.cpp:
```diff
@@ -126,7 +126,7 @@ void ApplicationCacheGroup::selectCache(LocalFrame& frame, const URL& passedMani
     ASSERT(frame.loader().documentLoader());
 
-    if (!frame.settings().offlineWebApplicationCacheEnabled())
+    if (frame.loader().documentLoader() && !frame.loader().documentLoader()->applicationCacheHost().isApplicationCacheEnabled())
```
It seems we don't need to null check documentLoader and this line should be moved to after:
`auto& documentLoader = *frame.loader().documentLoader();`

youennf replied:
This code will be removed soon, I prefer to keep it safe for now.

szewai commented on Source/WebCore/loader/appcache/ApplicationCacheGroup.cpp:
```diff
@@ -202,7 +202,7 @@ void ApplicationCacheGroup::selectCache(LocalFrame& frame, const URL& passedMani
 void ApplicationCacheGroup::selectCacheWithoutManifestURL(LocalFrame& frame)
 {
-    if (!frame.settings().offlineWebApplicationCacheEnabled())
+    if (frame.loader().documentLoader() && !frame.loader().documentLoader()->applicationCacheHost().isApplicationCacheEnabled())
```
Ditto; this should probably be moved to after `auto& documentLoader = *frame.loader().documentLoader();`

youennf replied:
I prefer being safe here and this code will be removed soon anyway.

szewai commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -1218,10 +1218,7 @@
-        (deny system-fcntl (with telemetry-backtrace) (appcache-fcntl-commands)))
+    (deny system-fcntl (with telemetry-backtrace) (appcache-fcntl-commands)))
 #else
     (allow system-fcntl (appcache-fcntl-commands))
```
If AppCache is disabled everywhere, shouldn't these be denied unconditionally?

youennf replied:
This seems best handled as a follow-up, I do not want to tighten the sandbox in this patch

szewai commented on Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
```diff
@@ -1683,10 +1683,7 @@
+    (deny system-fcntl (appcache-fcntl-commands))
 #else
     (allow system-fcntl (appcache-fcntl-commands))
 #endif
```
If AppCache is disabled everywhere, shouldn't the calls be denied unconditionally?

youennf replied:
Probably, but I prefer not further tightening the sandbox in the case of !HAVE(SANDBOX_STATE_FLAGS) in this patch.
This should probably be done in a follow-up

szewai commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/LocalStorageClear.mm:
```diff
@@ -90,118 +90,8 @@ - (void)userContentController:(WKUserContentController *)userContentController d
-#endif
-}
-
-TEST(WKWebView, ClearAppCache)
```
This test and database files should probably be kept, as we will need to verify the remaining AppCache data on disk can be removed via API.

youennf replied:
Right, this test should be removed when we remove the NetworkProcessProxy related code, which should be done in a follow-up, I will revert.

youennf replied:
As of AppCache data, given the time it has been obsolete, I would hope we do not need to keep the NetworkProcessProxy related code around

szewai commented on Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
```diff
@@ -1148,12 +1148,12 @@ - (void)setLocalFileContentSniffingEnabled:(BOOL)flag
 - (void)setOfflineWebApplicationCacheEnabled:(BOOL)flag
 {
-    [self _setBoolValue:flag forKey:WebKitOfflineWebApplicationCacheEnabledPreferenceKey];
+    // NO-OP
```
We probably don't need the comment as it's obvious

szewai commented on Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
```diff
@@ -1797,13 +1797,7 @@ void webkit_settings_set_enable_offline_web_application_cache(WebKitSettings* se
-
-    priv->preferences->setOfflineWebApplicationCacheEnabled(enabled);
-    g_object_notify_by_pspec(G_OBJECT(settings), sObjProperties[PROP_ENABLE_OFFLINE_WEB_APPLICATION_CACHE]);
+    // NO OP
```
We probably don't need this comment as no-op is obvious; `//Deprecated` may be more useful.
And I think these changes may need to be reviewed by glib reviewers, as I am not sure if they currently have clients that depend on these APIs, maybe @carlosgcampos @mcatanzaro?

mcatanzaro replied:
You did the right thing: keep the APIs, but make them no-ops. I doubt anything relies on this feature.

Please mark them as deprecated, though:

 * Add `Deprecated: 2.44` to the bottom of the doc comments on these functions.
 * Change `WEBKIT_API` to `WEBKIT_DEPRECATED` on the declarations in WebKitSettings.h.in
 * In  webkit_settings_class_init(), find `WebKitSettings:enable-offline-web-application-cache`, add `Deprecated: 2.44` and change the final argument to `static_cast<GParamFlags>(readWriteConstructParamFlags | G_PARAM_DEPRECATED)`

szewai commented on Source/WebKit/UIProcess/API/Cocoa/WKPreferencesPrivate.h:
```diff
@@ -103,8 +103,7 @@ typedef NS_ENUM(NSInteger, _WKPitchCorrectionAlgorithm) {
 
-// FIXME: This should be configured on the WKWebsiteDataStore.
-// FIXME: This property should not have the verb "is" in it.
+// FIXME: This property is deprecated, we should remove it.
```
We could move these to under `@interface WKPreferences (WKPrivateDeprecated)`?

youennf replied:
ok

szewai commented on Source/WebKit/UIProcess/API/Cocoa/WKPreferences.mm:
```diff
@@ -284,12 +284,12 @@ - (void)_setStorageBlockingPolicy:(_WKStorageBlockingPolicy)policy
 - (void)_setOfflineApplicationCacheIsEnabled:(BOOL)offlineApplicationCacheIsEnabled
 {
-    _preferences->setOfflineWebApplicationCacheEnabled(offlineApplicationCacheIsEnabled);
+    // NO-OP
```
The comment is unnecessary.

szewai commented on Source/WebKit/UIProcess/API/C/WKPreferences.cpp:
```diff
@@ -157,14 +157,14 @@ bool WKPreferencesGetLoadsImagesAutomatically(WKPreferencesRef preferencesRef)
+void WKPreferencesSetOfflineWebApplicationCacheEnabled(WKPreferencesRef, bool)
 {
-    toImpl(preferencesRef)->setOfflineWebApplicationCacheEnabled(offlineWebApplicationCacheEnabled);
+    // NO-OP
```
The comment is unnecessary.

mcatanzaro commented on Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitSettings.cpp:
```diff
@@ -63,11 +63,6 @@ static void testWebKitSettings(Test*, gconstpointer)
-    // Offline application cache is true by default.
-    g_assert_true(webkit_settings_get_enable_offline_web_application_cache(settings));
-    webkit_settings_set_enable_offline_web_application_cache(settings, FALSE);
-    g_assert_false(webkit_settings_get_enable_offline_web_application_cache(settings));
```
Instead of removing this test, please test that the default value is FALSE, then set it to TRUE, then confirm that it still returns FALSE (because the underlying setting has been removed).

youennf replied:
Will do.

mcatanzaro commented on Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
```diff
@@ -1777,33 +1774,32 @@ void webkit_settings_set_load_icons_ignoring_image_load_setting(WebKitSettings*
     g_return_val_if_fail(WEBKIT_IS_SETTINGS(settings), FALSE);
 
-    return settings->priv->preferences->offlineWebApplicationCacheEnabled();
+    return false;
```
```suggestion
    return FALSE;
```

mcatanzaro reviewed (commented):
WPE/GTK changes look good.

pvollan commented on Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
```diff
@@ -1909,8 +1906,6 @@
 (with-filter (require-all (require-not (lockdown-mode)) (require-not (state-flag "BlockIOKitInWebContentSandbox")))
     (allow syscall-unix (syscall-unix-downlevels-blocked-in-lockdown-mode)))
-(with-filter (require-not (state-flag "AppCacheDisabled"))
-     (allow syscall-unix (syscall-appcache)))
```
You may be able to remove the syscall-appcache list, too.

pvollan commented on Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
```diff
@@ -1683,10 +1683,7 @@
-    (with-filter (require-not (state-flag "AppCacheDisabled"))
-        (allow system-fcntl (appcache-fcntl-commands)))
-    (with-filter (state-flag "AppCacheDisabled")
-        (deny system-fcntl (appcache-fcntl-commands)))
+    (deny system-fcntl (appcache-fcntl-commands))
 #else
     (allow system-fcntl (appcache-fcntl-commands))
 #endif
```
If we are disabling AppCache on all OS versions, I think this could be removed along with `appcache-fcntl-commands`.

youennf replied:
I prefer doing this as a follow-up since this could potentially touch macOS downlevels.

annevk commented on Source/WebKit/UIProcess/API/C/WKPreferencesRef.h:
```diff
@@ -79,7 +79,7 @@ WK_EXPORT bool WKPreferencesGetLoadsImagesAutomatically(WKPreferencesRef prefere
-// Defaults to true.
+// Obsolete: no-op, defaults to false.
 WK_EXPORT void WKPreferencesSetOfflineWebApplicationCacheEnabled(WKPreferencesRef preferences, bool offlineWebApplicationCacheEnabled);
 WK_EXPORT bool WKPreferencesGetOfflineWebApplicationCacheEnabled(WKPreferencesRef preferences);
```
You should probably adopt WK_C_API_DEPRECATED here.

annevk commented on Source/WebKit/UIProcess/API/C/WKPreferences.cpp:
```diff
@@ -157,14 +157,13 @@ bool WKPreferencesGetLoadsImagesAutomatically(WKPreferencesRef preferencesRef)
 {
-    return toImpl(preferencesRef)->offlineWebApplicationCacheEnabled();
+    return false;
 }
```
I think technically these should be moved to the end of the file where the deprecated getters and setters are kept.

annevk commented on Source/WebKit/UIProcess/API/Cocoa/WKPreferencesPrivate.h:
```diff
@@ -247,6 +244,8 @@ typedef NS_ENUM(NSInteger, _WKPitchCorrectionAlgorithm) {
 #endif
 @property (nonatomic, setter=_setDisplayListDrawingEnabled:) BOOL _displayListDrawingEnabled WK_API_DEPRECATED("Display list drawing is no longer supported", macos(10.12, WK_MAC_TBA), ios(10.0, WK_IOS_TBA));
+// FIXME: We should remove this property.
+@property (nonatomic, setter=_setOfflineApplicationCacheIsEnabled:) BOOL _offlineApplicationCacheIsEnabled;
```
I thought it was unlikely we could remove API. This should probably adopt WK_API_DEPRECATED.

annevk commented on Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
```diff
@@ -1148,12 +1148,11 @@ - (void)setLocalFileContentSniffingEnabled:(BOOL)flag
 
 - (BOOL)offlineWebApplicationCacheEnabled
```
These should be moved into WebPrivateDeprecated.

annevk reviewed (commented):
Some nits on how to properly deprecate the preference (based on my experience of removing a lot of preferences).

Thanks for pushing this forward!

youennf committed:
Disable AppCache everywhere
https://bugs.webkit.org/show_bug.cgi?id=259783
rdar://113343269

Reviewed by Sihui Liu.

We remove the offline app cache runtime flag.
We disable using app cache, although we still expose DOMApplicationCache API based on the existing quirk.
We remove the tests and a follow-up will remove the implementation.

* LayoutTests/TestExpectations:
* LayoutTests/fast/dom/dom-window-property-gc-after-frame-detach-expected.txt:
* LayoutTests/fast/dom/dom-window-property-gc-after-frame-detach.html:
* LayoutTests/fast/dom/event-handler-attributes.html:
* LayoutTests/fast/events/destroyed-atomic-string.html:
* LayoutTests/fast/events/dispatchEvent-crash-expected.txt:
* LayoutTests/fast/events/dispatchEvent-crash.html:
* LayoutTests/fast/loader/window-properties-restored-from-page-cache-expected.txt:
* LayoutTests/fast/loader/window-properties-restored-from-page-cache.html:
* LayoutTests/http/tests/appcache/: Removed.
* LayoutTests/http/tests/contentfiltering/load-substitute-data-from-appcache-expected.txt: Removed.
* LayoutTests/http/tests/contentfiltering/load-substitute-data-from-appcache.html: Removed.
* LayoutTests/http/tests/contentfiltering/resources/appcache.html: Removed.
* LayoutTests/http/tests/dom/cross-origin-detached-window-properties-expected.txt:
* LayoutTests/http/tests/dom/cross-origin-detached-window-properties.html:
* LayoutTests/http/tests/dom/same-origin-detached-window-properties-expected.txt:
* LayoutTests/http/tests/dom/same-origin-detached-window-properties.html:
* LayoutTests/http/tests/security/appcache-in-private-browsing-expected.txt: Removed.
* LayoutTests/http/tests/security/appcache-in-private-browsing.html: Removed.
* LayoutTests/http/tests/security/appcache-switching-private-browsing-expected.txt: Removed.
* LayoutTests/http/tests/security/appcache-switching-private-browsing.html: Removed.
* LayoutTests/http/tests/security/cross-origin-appcache-allowed-expected.txt: Removed.
* LayoutTests/http/tests/security/cross-origin-appcache-allowed.html: Removed.
* LayoutTests/http/tests/security/cross-origin-appcache-expected.txt: Removed.
* LayoutTests/http/tests/security/cross-origin-appcache.html: Removed.
* LayoutTests/http/tests/security/resources/cross-origin-iframe-for-appcache-allowed.html: Removed.
* LayoutTests/http/tests/security/resources/cross-origin-iframe-for-appcache.html: Removed.
* LayoutTests/http/tests/security/resources/frame-for-appcache-in-private-browsing.html: Removed.
* LayoutTests/http/tests/security/resources/same-origin-iframe-for-appcache-blocked.html: Removed.
* LayoutTests/http/tests/security/same-origin-appcache-blocked-expected.txt: Removed.
* LayoutTests/http/tests/security/same-origin-appcache-blocked.html: Removed.
* LayoutTests/http/tests/ssl/resources/origin-usage-iframe-1.html: Removed.
* LayoutTests/http/tests/ssl/resources/origin-usage-iframe-2.html: Removed.
* LayoutTests/http/tests/ssl/upgrade-origin-usage-expected.txt: Removed.
* LayoutTests/http/tests/ssl/upgrade-origin-usage.html: Removed.
* LayoutTests/js/frame-application-cache-with-listener-crash.html: Removed.
* LayoutTests/platform/ios-wk2/TestExpectations:
* LayoutTests/platform/mac-monterey-wk1/TestExpectations:
* LayoutTests/platform/mac-monterey-wk1/http/tests/appcache/remove-cache-expected.txt: Removed.
* LayoutTests/platform/mac-monterey/TestExpectations:
* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/platform/mac-ventura/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/wincairo/TestExpectations:
* LayoutTests/platform/wk2/TestExpectations:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.cpp:
(WTF::computeSDKAlignedBehaviors):
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WebCore/loader/appcache/ApplicationCacheGroup.cpp:
(WebCore::ApplicationCacheGroup::selectCache):
(WebCore::ApplicationCacheGroup::selectCacheWithoutManifestURL):
* Source/WebCore/loader/appcache/ApplicationCacheHost.cpp:
(WebCore::ApplicationCacheHost::isApplicationCacheEnabled):
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldEnableApplicationCacheQuirk const):
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultOfflineWebApplicationCacheEnabled): Deleted.
* Source/WebKit/Shared/WebPreferencesDefaultValues.h:
* Source/WebKit/UIProcess/API/C/WKPreferences.cpp:
(WKPreferencesSetOfflineWebApplicationCacheEnabled):
(WKPreferencesGetOfflineWebApplicationCacheEnabled):
* Source/WebKit/UIProcess/API/C/WKPreferencesRef.h:
* Source/WebKit/UIProcess/API/Cocoa/WKPreferences.mm:
(-[WKPreferences _offlineApplicationCacheIsEnabled]):
(-[WKPreferences _setOfflineApplicationCacheIsEnabled:]):
* Source/WebKit/UIProcess/API/Cocoa/WKPreferencesPrivate.h:
* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webkit_settings_class_init):
(webkit_settings_get_enable_offline_web_application_cache):
(webkit_settings_set_enable_offline_web_application_cache):
* Source/WebKit/UIProcess/API/glib/WebKitSettings.h.in:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_historyItemClient):
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
* Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
(-[WebPreferences offlineWebApplicationCacheEnabled]):
(-[WebPreferences setOfflineWebApplicationCacheEnabled:]):
* Tools/DumpRenderTree/TestOptions.cpp:
(WTR::TestOptions::defaults):
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKit/WKPreferences.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsiteDataStoreCustomPaths.mm:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitSettings.cpp:
(testWebKitSettings):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebsiteData.cpp:
(serverCallback):
(testWebsiteDataConfiguration):
(testWebsiteDataEphemeral):
(beforeAll):
(testWebsiteDataAppcache): Deleted.
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):

Originally-landed-as: 39f334671a68. https://bugs.webkit.org/show_bug.cgi?id=259783
Canonical link: https://commits.webkit.org/273297@main

pvollan commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -1218,10 +1218,7 @@
-        (allow system-fcntl (appcache-fcntl-commands)))
-    (with-filter (state-flag "AppCacheDisabled")
-        (deny system-fcntl (with telemetry-backtrace) (appcache-fcntl-commands)))
+    (deny system-fcntl (with telemetry-backtrace) (appcache-fcntl-commands)))
```
I think there is one too many closing parenthesis here.
